Speedup the code with some minor changes and a bit of cleanup

EarthScientist opened the pull request:
This PR will speedup the codebase significantly and runs the same way as you did before.

It takes ~3mins to run across all files.

EarthScientist commented:
directly related to #1

julienschroder commented:
Interesting, so the change of folder + autoclose + looking up the coordinate outside the function (which makes sense) all are making a huge difference!
Thanks merging now.

EarthScientist commented:
you had some memory leaks, and were using the directory with the _really_
large files.  Also there was some computation that was being done that was
not being used.  So I removed that stuff, changed around the computation
order, and this is what the result is.  So it is essentially all your code,
I just spruced it up a little bit! :)

Give it a try and see how fast it runs on atlas!

Michael



On Wed, Dec 6, 2017 at 12:11 AM, jschroder <notifications@github.com> wrote:

> Merged #2 <https://github.com/julienschroder/DOD_WRF/pull/2>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/julienschroder/DOD_WRF/pull/2#event-1374361926>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABUarI1_cJ0x9ePOVRoKK0F3EjvJq39bks5s9lo4gaJpZM4Q3lKT>
> .
>
